### PR TITLE
Document the methods and properties under com.ibm.panel interface

### PR DESCRIPTION
**Pull request by PriyangaRamasamy**
Provides documentation on methods and properties present under com.ibm.panel interface hosted by com.ibm.PanelApp service.

### CODE REVIEW

**Comment by PriyangaRamasamy** [comment]
jenkins run tests please

File: yaml/com/ibm/Panel.interface.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+description: >
+    Implements control panel activities.
+methods:
+    - name: Display
+      description: >
+          Commands the op-panel to display the given lines.
```
Comment (SunnySrivastava1984) [inline yaml/com/ibm/Panel.interface.yaml:6]: Reword?: API to display data on LCD panel. 
Since there are two panels, its will be handy if we explicitly tell that.

File: yaml/com/ibm/Panel.interface.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            description: >
+                First line that needs to be displayed in op-panel LCD.
+          - name: line2
+            type: string
+            description: >
+                Second line that needs to be displayed in op-panel LCD.
+
+    - name: TriggerPanelLampTest
+      description: >
+          Commands the op-panel to trigger panel lamp test.
```
Comment (SunnySrivastava1984) [inline yaml/com/ibm/Panel.interface.yaml:19]: Reword? API to trigger lamp test from panel.

"to trigger panel lamp test" appears as if there is a lamp test for panel.

File: yaml/com/ibm/Panel.interface.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      parameters:
+          - name: event_id
+            type: int32
+            description: >
+                Event ID = 0 simulates UP button. Event ID = 1 simulates DOWN
+                button. Event ID = 2 simulates ENTER button.
+
+    - name: ExecuteFunction
+      description: >
+          Trigger panel function directly.
```
Comment (SunnySrivastava1984) [inline yaml/com/ibm/Panel.interface.yaml:48]: Reword? API to trigger panel function.

Directly does not make any sense here.

File: yaml/com/ibm/Panel.interface.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      parameters:
+          - name: function_number
+            type: byte
+            description: >
+                Execute the given function number. Supports to trigger only IBMi
+                functions {21, 22, 34, 65, 66, 67, 68, 69, 70}.
```
Comment (SunnySrivastava1984) [inline yaml/com/ibm/Panel.interface.yaml:54]: "Execute the given function number" is fine here.
Remove function numbers and IBMi from here.
As later we can add more functions to this and we don't want to come back and modify interface file every time.

File: yaml/com/ibm/Panel.interface.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      returns:
+          - name: output
+            type: struct[boolean, string, string]
+            description: >
+                Tuple with values return code, panel display line1, panel
```
Comment (SunnySrivastava1984) [inline yaml/com/ibm/Panel.interface.yaml:59]: Type you have already explained in "type:" section.

File: yaml/com/ibm/Panel.interface.yaml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+          - xyz.openbmc_project.Common.Error.NotAllowed
+
+    - name: getEnabledFunctions
+      description: >
+          Get list of enabled functions.
+      returns:
+          - name: enabled functions list
+            type: array[byte]
+            description: >
+                List of enabled functions from the set of IBMi functions list
```
Comment (SunnySrivastava1984) [inline yaml/com/ibm/Panel.interface.yaml:72]: Same here, get rid of IBMi and function numbers.
Just "List of enabled functions" is sufficient.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: yaml/com/ibm/Panel.interface.yaml (hunk 1 above)
Comment (PriyangaRamasamy) [reply yaml/com/ibm/Panel.interface.yaml]: done

File: yaml/com/ibm/Panel.interface.yaml (hunk 5 above)
Comment (SunnySrivastava1984) [reply yaml/com/ibm/Panel.interface.yaml:60]: I was mentioning the word Tuple.Anyways I am ok with this as well.

File: yaml/com/ibm/Panel.interface.yaml (hunk 6 above)
Comment (SunnySrivastava1984) [reply yaml/com/ibm/Panel.interface.yaml:73]: Generic work of this function is to return a list of enabled functions. In future if there is a requirement to add more functions to this other than IBMi then we can do that. We will not have to modify this document.

Comment (SunnySrivastava1984) [reply yaml/com/ibm/Panel.interface.yaml:73]: As your function name nowhere suggests it is limited to IBMi function, similarly its description should not.

Comment (PriyangaRamasamy) [reply yaml/com/ibm/Panel.interface.yaml:73]: We can't express the whole intent in the function name itself. thats why we have a document to describe the function. From the description people can know what this function does. whenever requirement changes, we can update it.
@jinuthomas ^ any thoughts on this?

Comment (SunnySrivastava1984) [reply yaml/com/ibm/Panel.interface.yaml:73]: There was a similar discussion at the time of function implementation as well? The whole target is to keep the things as generic as possible.
https://github.com/ibm-openbmc/ibm-panel/pull/190#discussion_r1265394738

File: yaml/com/ibm/Panel.interface.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
+description: >
+    Implements control panel activities.
+methods:
+    - name: TriggerPanelLampTest
+      description: >
+          API to trigger lamp test on panel.
```
Comment (jinuthomas) [inline yaml/com/ibm/Panel.interface.yaml:6]: Method to

File: yaml/com/ibm/Panel.interface.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+      parameters:
+          - name: status
+            type: boolean
+            description: >
+                Panel lamp test request.
```
Comment (jinuthomas) [inline yaml/com/ibm/Panel.interface.yaml:11]: This is description for the parameter?

File: yaml/com/ibm/Panel.interface.yaml [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+    - name: toggleFunctionState
+      description: >
+          Toggles the state of given panel functions.
```
Comment (jinuthomas) [inline yaml/com/ibm/Panel.interface.yaml:15]: I would avoid this function as this function is used by PLDM to indicate to the panel that a function has changed in enablement state

File: yaml/com/ibm/Panel.interface.yaml [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+      parameters:
+          - name: bitmap
+            type: array[byte]
+            description: >
+                Bit number in bitmap will have one to one mapping with function
+                number. Bit value 1 to enable and 0 to disable the function.
+
+    - name: ExecuteFunction
+      description: >
+          API which can execute the function that is returned by
```
Comment (jinuthomas) [inline yaml/com/ibm/Panel.interface.yaml:25]: Method to

File: yaml/com/ibm/Panel.interface.yaml [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+          getEnabledFunctions.
+      parameters:
+          - name: function_number
+            type: byte
+            description: >
+                One of the function numbers returned by getEnabledFunctions.
```
Comment (jinuthomas) [inline yaml/com/ibm/Panel.interface.yaml:31]: don't have to specify here that it should be from getEnabledFunctions.

File: yaml/com/ibm/Panel.interface.yaml [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+            description: >
+                Tuple with values return code, panel display line1, panel
+                display line2, if the execution is successful.
+      errors:
+          - xyz.openbmc_project.Common.Error.InternalFailure
+          - xyz.openbmc_project.Common.Error.NotAllowed
+
+    - name: getEnabledFunctions
+      description: >
+          Provides the list of enabled functions which can be triggerred
```
Comment (jinuthomas) [inline yaml/com/ibm/Panel.interface.yaml:44]: Method to

File: yaml/com/ibm/Panel.interface.yaml [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+                other apps.
+
+properties:
+    - name: ACFWindowActive
+      type: boolean
+      description: >
+          Property which tells if the window to upload ACF is enabled or
+          disabled.
+
+    - name: OSIPLMode
```
Comment (jinuthomas) [inline yaml/com/ibm/Panel.interface.yaml:60]: why is this needed ?

File: yaml/com/ibm/Panel.interface.yaml (hunk 5 above)
Comment (jinuthomas) [reply yaml/com/ibm/Panel.interface.yaml:36]: would like the line rewritten . why here tuple and above struct ?

File: gen/com/ibm/meson.build [hunk 14]
```diff
@@ -5,6 +5,21 @@ subdir('Host')
 subdir('License')
 subdir('Logging')
 subdir('PLDM')
+subdir('Panel')
+generated_others += custom_target(
```
Comment (jinuthomas) [inline gen/com/ibm/meson.build:9]: why is this required ?

**Review by jinuthomas — CHANGES_REQUESTED** [review]

File: yaml/com/ibm/Panel.interface.yaml (hunk 8 above)
Comment (PriyangaRamasamy) [reply yaml/com/ibm/Panel.interface.yaml]: yes.. modified the statement

File: yaml/com/ibm/Panel.interface.yaml (hunk 12 above)
Comment (PriyangaRamasamy) [reply yaml/com/ibm/Panel.interface.yaml]: done

**Review by SunnySrivastava1984 — APPROVED** [review]

**Review by jinuthomas — APPROVED** [review]

**Review by jinuthomas — APPROVED** [review]
